### PR TITLE
docs: Update issue templates and release resources

### PR DESCRIPTION
**Pull request by aarnq**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [x] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

This updates the issue templates and release resources, including the new public QA checklist for Apps.

I also updated the pre-commit hooks, and accidentally let all workflows run through prettier.

- Part of https://github.com/elastisys/ck8s-qa/issues/3

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by crssnd — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
+<!-- Choose your PR title carefully as it will be used as the entry in the changelog! -->
 
 > [!warning]
-> **This is public repository, ensure not to disclose:**
-> - [ ] personal data beyond what is necessary for interacting with this pull request
-> - [ ] business confidential information, such as customer names
+> **This is a public repository, ensure not to disclose:**
+>
+> - [ ] personal data beyond what is necessary for interacting with this pull request, nor
+> - [ ] business confidential information, such as customer names.
```
Comment (davidumea) [inline .github/pull_request_template.md:7]: This is what I was referring to in the other comment, shouldn't it be the same in issues and PRs?

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -1,54 +1,58 @@
...
+  - type: markdown
+    attributes:
+      value: |-
+        Thank you for taking the time to help improve Compliant Kubernetes!
+
+        > [!warning]
+        > **This is a public repository, ensure not to disclose:**
+        >
+        > - [x] personal data beyond what is necessary for interacting with this issue, nor
+        > - [x] business confidential information, such as customer names.
```
[the remark this answers is not in the document]
Comment (aarnq) [reply .github/ISSUE_TEMPLATE/bug-report.yml:14]: In issue forms the markdown section is rendered into when the issue is created, and only visible to the author when it is being written. Once it is created it is removed, and the author cannot edit this message.

I feel like this was a "less cluttered" version of it as it is most critical for the issue author to have that warning, and it makes less sense for anyone that later picks it up (as the discussion should already be general enough in the issue to not lead into highly specific instances).

Let me know if that makes sense or not. :sweat_smile:

File: release/README.md [hunk 3]
```diff
@@ -10,49 +10,65 @@ The following constraints apply on releases:
...
 
 1. Patch releases (vX.Y.**Z**):
 
     > *Patches to fix known vulnerabilities threatening user data assessed as an immediate risk.*
 
+    - May **only** perform patch additions or modifications to the application and command line interfaces.
+
+      The interfaces **must** keep similar behaviour between different patch releases to ensure that they work with the same sets of features and tools, and can be applied over any patch version on the same minor version.
+
+      Example of allowed patches would be to correct failty behaviour.
```
Comment (davidumea) [inline release/README.md:55]:
```suggestion
      Example of allowed patches would be to correct faulty behaviour.
```

**Review by davidumea — APPROVED** [review]

**Comment by anders-elastisys** [comment]
@aarnq could we add a step for testing harbor restore from a backup made on the current release as well? Currently we have a test for restoring from a previous backup, but recently we noticed an issue with harbor restores and to avoid such issues it might be good to test restoring backups from both the previous and current apps version.

**Comment by aarnq** [comment]
> @aarnq could we add a step for testing harbor restore from a backup made on the current release as well? Currently we have a test for restoring from a previous backup, but recently we noticed an issue with harbor restores and to avoid such issues it might be good to test restoring backups from both the previous and current apps version.

Might be good to do in general for backup solutions that we manage. :+1:

**Pull request merged**